providers/provider-Bandwidth.php

```diff
@@ -0,0 +1,202 @@
+                'default'   => '',
+                'required'  => false,
+                'placeholder' => _('Enter Callback User ID (optional)'),
+            ),
+            'callback_password' => array(
+                'type'      => 'string',
+                'label'     => _('Callback Password'),
+                'help'      => _("Enter the Callback Password (optional)"),
+                'default'   => '',
+                'required'  => false,
+                'class'     => 'confidential',
+                'placeholder' => _('Enter Callback Password (optional)'),
+            )
+        );
+    }
+
+    public function sendMedia($id, $to, $from, $message = null)
+    {
+        $req = array(
+            "applicationId" => $this->getConfig($this->nameRaw)['application_id'],
+            "to"    => array($to),
+            "from"  => $from,
+            "media" => $this->media_urls($id)
+        );
+        if ($message) {
+            $attr['text'] = $message;
+        }
+
+        $this->sendBandwidth($req, $id);
+        return true;
+    }
+
+    public function sendMessage($id, $to, $from, $message = null)
+    {
+        $req = array(
+            "applicationId" => $this->getConfig($this->nameRaw)['application_id'],
+            "to"    => array($to),
+            "from"  => $from,
+            "text"  => $message
+        );
+        $this->SendBandwidth($req, $id);
+        return true;
+    }
+
+    private function sendBandwidth($payload, $mid)
+    {
+        $config = $this->getConfig($this->nameRaw);
+
+        $headers = array(
+            "Content-Type" => "application/json",
+            "Authorization" => "Basic " . base64_encode($config['api_token'] . ":" . $config['api_secret'])
+        );
+        $url = "https://messaging.bandwidth.com/api/v2/users/" . $config['account_id'] . "/messages";
+        $json = json_encode($payload);
+        $session = \FreePBX::Curl()->requests($url);
+
+        try {
+            $bandwidthResponse = $session->post('', $headers, $json, array());
+            freepbx_log(FPBX_LOG_INFO, sprintf("%s responds: HTTP %s, %s", $this->nameRaw, $bandwidthResponse->status_code, $bandwidthResponse->body), true);
+
+            if (!$bandwidthResponse->success) {
+                throw new \Exception("HTTP $bandwidthResponse->status_code, $bandwidthResponse->body");
+            }
+            $this->setDelivered($mid);
+        } catch (\Exception $e) {
+            throw new \Exception('Unable to send message: ' . $e->getMessage());
+        }
+    }
+
+    public function callPublic($connector)
+    {
+        $config = $this->getConfig($this->nameRaw);
+
+        // Check if callback authentication is enabled/expected
+        // If it is, check that the user and password match the values coming from bandwidth
+        if ($config['callback_user_id'] && $config['callback_password']) {
+            
+            $user = $_SERVER['PHP_AUTH_USER'];
```

Comment:
`$user = (empty($_SERVER['PHP_AUTH_USER']) ? '' : $_SERVER['PHP_AUTH_USER'];` 

same for $password assignment, again to avoid exception in PHP8 for referencing unassigned array keys.